lib/Doctrine/DBAL/Platforms/DB2Platform.php

```diff
@@ -878,4 +881,57 @@ protected function getReservedKeywordsClass()
     {
         return Keywords\DB2Keywords::class;
     }
+    
+    /**
+     * Prepare ORDER BY string for OVER() if applicable
+     * 
+     * @param  string $query
+     * 
+     * @return string
+     */
+    private function getOrderByForOver($query)
+    {
+        //preset empty array
+        $orderByArray = [];
+
+        // remove extra white space
+        $query = preg_replace('/\s+/', ' ', $query);
+
+        //determine if 'ORDER BY' is part of the query
+        $orderByFound = preg_match('/(\bORDER\b)(\s*)(\bBY\b)(?!.*\b\1\b)/i', $query, $orderByPositionArray, PREG_OFFSET_CAPTURE);
+
+        // ORDER BY found in query string
+        if (0 !== $orderByFound) {
+            $orderByPosition = $orderByPositionArray[0][1];
+
+            $queryArray = preg_split('/[, ]/', substr($query, 0, $orderByPosition -1));
+            $orderByArray = explode(',', substr($query, $orderByPosition + strlen('ORDER BY')));
```

Comment:
`strlen('ORDER BY')` will probably corrupt the query if there's more than one space between `ORDER` and `BY` in the original query.

Reply:
I think line 898 should take care of the whitespace, which should simplify the whole preg_split and explode madness.

All I tried to do is help with the IBM issue, but you are worth in code review than I am! btw. That's a compliment 😆 

Anyway, maybe with some help we can get than show on the road; Alan, Clark... you are more than welcome in joining me...